### PR TITLE
Ran rustfmt and added formatting checks to CI

### PR DESCRIPTION
**Pull request by Kixiron**
* Ran rustfmt across the codebase
* Added rustfmt checks to the github ci

### CODE REVIEW

File: tests/reftest/mod.rs [hunk 1]
```diff
@@ -19,9 +19,14 @@ fn reftest() {
...
     reftest_scaled_file(&base.join("rgb.jpg"), 500, 333, &base.join("rgb.png"));
-    reftest_scaled_file(&base.join("rgb.jpg"), 250, 167, &base.join("rgb_250x167.png"));
-    reftest_scaled_file(&base.join("rgb.jpg"), 125, 84,  &base.join("rgb_125x84.png"));
-    reftest_scaled_file(&base.join("rgb.jpg"), 63,  42,  &base.join("rgb_63x42.png"));
+    reftest_scaled_file(
+        &base.join("rgb.jpg"),
+        250,
+        167,
+        &base.join("rgb_250x167.png"),
+    );
```
Comment (HeroicKatora) [inline tests/reftest/mod.rs:27]: There's must be better algorithm for consistent fmt, I'm sure. We just haven't found it yet. But I shall accept it :joy:

File: src/huffman.rs [hunk 2]
```diff
@@ -292,55 +306,99 @@ fn derive_huffman_codes(bits: &[u8; 16]) -> Result<(Vec<u16>, Vec<u8>)> {
...
     if dc_huffman_tables[0].is_none() && scan.dc_table_indices.iter().any(|&i| i == 0) {
         // Table K.3
-        dc_huffman_tables[0] = Some(HuffmanTable::new(
-            &[0x00, 0x01, 0x05, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00],
-            &[0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B], HuffmanTableClass::DC).unwrap());
+        dc_huffman_tables[0] = Some(
+            HuffmanTable::new(
+                &[
+                    0x00, 0x01, 0x05, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00,
+                    0x00, 0x00, 0x00,
```
[the remark this answers is not in the document]
Comment (Kixiron) [reply src/huffman.rs]: I addressed it, is it better now?

**Comment by kornelski** [comment]
I think rustfmt checks in CI are counter-productive. It slows down CI feedback, and rejects PRs for trivial reasons.

But the worst effect of `rustfmt` is blowing up small code changes into big reformatting of entire blocks. It makes reviewing of PRs more difficult. Addition or deletion of a single line or word has a nice readable diff, but if it happens to push the code over some threshold in `rustfmt`, it can cause entire block to be rewrapped, and then it's harder to see what has actually changed.

**Pull request closed without merging**